app/src/main/java/com/celzero/bravedns/service/ConnectionMonitor.kt

```diff
@@ -0,0 +1,264 @@
+        if(DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - onLost")
+        if (!isAllNetworkSupported) {
+            val message = constructMessage(MSG_ADD_ACTIVE_NETWORK)
+            serviceHandler?.sendMessage(message)
+        } else {
+            val message = constructMessage(MSG_ADD_ALL_NETWORKS)
+            serviceHandler?.sendMessage(message)
+        }
+    }
+
+    override fun onCapabilitiesChanged(network: Network, networkCapabilities: NetworkCapabilities) {
+        serviceHandler?.removeMessages(MSG_ADD_ACTIVE_NETWORK, null)
+        serviceHandler?.removeMessages(MSG_ADD_ALL_NETWORKS, null)
+        if(DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - onCapabilitiesChanged")
+        if (!isAllNetworkSupported) {
+            val message = constructMessage(serviceHandler?.obtainMessage(), MSG_ADD_ACTIVE_NETWORK)
+            serviceHandler?.sendMessage(message)
+        } else {
+            val message = constructMessage(serviceHandler?.obtainMessage(), MSG_ADD_ALL_NETWORKS)
+            serviceHandler?.sendMessage(message)
+        }
+    }
+
+    override fun onLinkPropertiesChanged(network: Network, linkProperties: LinkProperties) {
+        serviceHandler?.removeMessages(MSG_ADD_ACTIVE_NETWORK, null)
+        serviceHandler?.removeMessages(MSG_ADD_ALL_NETWORKS, null)
+        if(DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - onLinkPropertiesChanged")
+        if (!isAllNetworkSupported) {
+            val message = constructMessage(serviceHandler?.obtainMessage(), MSG_ADD_ACTIVE_NETWORK)
+            serviceHandler?.sendMessage(message)
+        } else {
+            val message = constructMessage(serviceHandler?.obtainMessage(), MSG_ADD_ALL_NETWORKS)
+            serviceHandler?.sendMessage(message)
+        }
+    }
+
+    /**
+     * Called when the preference(isAddAllNetwork) is changed.
+     * if the preference is true, add only active network
+     * else, add all available network.
+     */
+    fun onPreferenceChanged(){
+        isAllNetworkSupported = persistentState.isAddAllNetwork
+        serviceHandler?.removeMessages(MSG_ADD_ACTIVE_NETWORK, null)
+        serviceHandler?.removeMessages(MSG_ADD_ALL_NETWORKS, null)
+        if (DEBUG) Log.d(LOG_TAG, "ConnectionMonitor - onPreferenceChanged")
+        if (!isAllNetworkSupported) {
+            val message = constructMessage(serviceHandler?.obtainMessage(), MSG_ADD_ACTIVE_NETWORK)
+            serviceHandler?.sendMessage(message)
+        } else {
+            val message = constructMessage(serviceHandler?.obtainMessage(), MSG_ADD_ALL_NETWORKS)
+            serviceHandler?.sendMessage(message)
+        }
+    }
+
+    fun getNetworkList(): MutableSet<Network> {
+        return currentNetworkList
+    }
+
+    /**
+     * Constructs the message object for Network handler.
+     * Add the active network to the message object
+     * in case of setUnderlying network has only active networks.
+     */
+    private fun constructMessage(what: Int): Message {
```

Comment:
Is this function used? If not remove.

Reply:
Removed the unused function.

Reply:
There were two constructMessage(). One of the constructMessage() is removed which had obtainMessage() as argument.